extensions/wikia/UserLogin/js/FacebookLogin.js

```diff
@@ -287,6 +287,30 @@
 					}
 				}
 			});
+
+			this.initSignupFormValidation();
+		},
+
+		initSignupFormValidation: function () {
+			var validator,
+				wikiaForm = this.signupForm.wikiaForm,
+				inputs = wikiaForm.inputs,
+				inputsToValidate = ['username', 'password'];
+
+			if (inputs.email) {
+				inputsToValidate.push('email');
+			}
+
+			validator = new UserSignupAjaxValidation({
+				wikiaForm: wikiaForm,
+				inputsToValidate: inputsToValidate,
+				submitButton: inputs.submit
+			});
+
+			// Add validation on blur event for all inputs to validate
+			inputsToValidate.forEach(function (inputName) {
+				inputs[inputName].on('blur', validator.validateMappedInput.bind(validator));
```

Comment:
Cool, yeah, logically that makes sense, so I did a test, and indeed, it is faster to bind once. http://jsperf.com/event-bindings-with-jquery. Will fix!